chore/check-javadoc-regressions.py

```diff
@@ -0,0 +1,327 @@
+            if regex.search(line):
+                print(line)
+
+        # If the user has a maven problem that prevents checkstyle from running they'd likely
+        # appreciate *some* output. If we can't extract a violation count the checkstyle plugin
+        # probably didn't successfully. Let's warn them.
+        if extract_violation_count(checkstyle_output) is None:
+            print()
+            print(warn("Checkstyle output doesn't contain a violation count. Maybe your setup has an error?"))
+            if len(sys.argv) > 1:
+                print(
+                    f" {hl('Suggestion')}: Run this script without arguments and inspect the unfiltered "
+                    "checkstyle output"
+                )
+
+
+def fix_colors_windows_cmd():
+    # Taken from https://github.com/Microsoft/WSL/issues/1173#issuecomment-459934080
+    import ctypes
+
+    kernel32 = ctypes.windll.kernel32
+    kernel32.SetConsoleMode(kernel32.GetStdHandle(-11), 7)
+
+
+def print_help():
+    print(f"""
+ {hl('USAGE')}
+   {sys.argv[0]} [COMPARE_WITH_MASTER | <regex file filter>] [--help]
+
+ {hl('DESCRIPTION')}
+   This is a small helper script to ensure the Javadoc quality in the project does not
+   deteriorate. It runs during CI and enforces that new changes never increase the amount
+   of checkstyle Javadoc errors.
+   Additionally, it tries to create a helpful output showing you the approximate location
+   of any errors you might introduce so you have some guidance while fixing them.
+
+ {hl('EXAMPLES')}
+   {success('Compare your branch with the master branch to check for regressions.')}
+     This will checkout the master branch, run checkstyle, checkout your branch
+     and compare the results.
+     Note that the checkout will {hl('fail')} if you have incompatible uncommitted changes.
+
+     {warn('python ' + sys.argv[0] + ' COMPARE_WITH_MASTER')}
+
+   {success('List all violations in a set of files in your current working directory.')}
+     Running this script with a regex as its only argument will execute checkstyle in
+     your working directory and filter the results using the regex you provided.
+     This can be useful if you want to find all violations in files you touched or
+     files the CI found errors in.
+
+     {warn('python ' + sys.argv[0])} src/main/java/
+
+   {success('List all violations in your current working directory.')}
+     Running this script without any arguments will echo the full maven checkstyle
+     output back to you.
+     This can be useful if you just want to invoke the maven checkstyle command
+     with an adequate configuration.
+
+     {warn('python ' + sys.argv[0])}
+    """[1:])
+
+
+if __name__ == "__main__":
+    if os.name == "nt":
+        fix_colors_windows_cmd()
+
+    if "--help" in sys.argv:
+        print_help()
+        exit(0)
+
+    if len(sys.argv) > 2:
+        print(f"Usage: {sys.argv[0]} [COMPARE_WITH_MASTER | regex file filter]")
```

Comment:
Perhaps also add `--help` here? Otherwise it'll be hard to know one can use it. This is why I almost always go for using `argparse` whenever I have a script that's in any way non-trivial :)

```suggestion
        print(f"Usage: {sys.argv[0]} [COMPARE_WITH_MASTER | regex file filter] [--help]")
```

Reply:
I was tempted to use it, but properly configuring it to reject invalid input here wasn't really trivial and the help page isn't what argparse would generate either (but hopefully easier to read, see image below). I was also clinging to the hope that the script was trivial, but the length of the conversations in the two issues might prove me wrong...

I will make it call `print_help()` in there, I think. Then the usage is consistent.

![grafik](https://user-images.githubusercontent.com/20284688/125072157-7ff16f00-e0ba-11eb-8f97-d436cd2fe635.png)